### PR TITLE
Wip/json performance

### PR DESCRIPTION
**Pull request by briot**
Performance improvements (28%) for GNATCOLL.JSON

### CODE REVIEW

**Comment by briot** [comment]
There hasn't been any followup to this pull request in a month. Are there issues with this patch ?

File: src/gnatcoll-json-utility.ads [hunk 1]
```diff
@@ -34,7 +34,7 @@ private package GNATCOLL.JSON.Utility is
    --  Translates an UTF-8 encoded unbounded string into a JSON-escaped string
 
    function Un_Escape_String
```
Comment (Nikokrock) [inline src/gnatcoll-json-utility.ads]: This is a public interface. You need to keep a version of the function that is backward compatible with the previous version

**Comment by briot** [comment]
> In src/gnatcoll-json-utility.ads:
> 
> > @@ -34,7 +34,7 @@ private package GNATCOLL.JSON.Utility is
>     --  Translates an UTF-8 encoded unbounded string into a JSON-escaped string
>  
>     function Un_Escape_String
> 
> This is a public interface. You need to keep a version of the function that is backward compatible with the previous version


It’s not really public, since it is in a private package.
It’s true that children of GNATCOLL.JSON might be impacted, but it isn’t expected that
users need to look under the hood here.

**Comment by briot** [comment]
superceded by #23

**Pull request closed without merging**